### PR TITLE
Replace print statements by calls to logging module

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #12

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Not sure what's going on here. Unable to reproduce locally

**Comment by MichaelClerx** [comment]
That seems to have fixed it

File: fc/protocol.py [hunk 1]
```diff
@@ -660,24 +670,3 @@ def get_path(self, base_path, path):
             new_path = os.path.join(library, path)
         return new_path
 
-    def set_indent_level(self, indent_level):
-        """Set the level of indentation to use for progress output."""
-        self.indent_level = indent_level
-
-    def log_progress(self, *args):
```
Comment (jonc125) [inline fc/protocol.py:667]: In getting rid of these helper methods, you're no longer indenting the log output from nested protocols, which makes the output harder to follow. Could we recover that functionality, or something equivalent?

File: fc/templates/weblab_model.pyx [hunk 2]
```diff
@@ -312,5 +299,5 @@ cdef class {{ class_name }}(CvodeSolver):
         """
         # TODO Update this (and rest of fc) to Python3
         # TODO Use logging here, or raise an exception
-        print >>sys.stderr, '  ' * self.indent_level, 'set_solver: Models implemented using Cython contain a built-in ODE solver, so ignoring setting.'
+        print >>sys.stderr, 'set_solver: Models implemented using Cython contain a built-in ODE solver, so ignoring setting.'
```
Comment (jonc125) [inline fc/templates/weblab_model.pyx:302]: Is there a reason for not using logging here?

File: test/test_simple_protocols.py [hunk 3]
```diff
@@ -6,21 +6,21 @@
...
 
-def test_core_post_proc_txt():
+def test_core_postproc():
     proto_file = 'test/protocols/test_core_postproc.txt'
     proto = fc.Protocol(proto_file)
     proto.run()
 
 
-def test_graph_txt():
+def test_graph_state():
```
Comment (jonc125) [inline test/test_simple_protocols.py:21]: This duplicates a test name in test_code_generation.py

File: fc/templates/weblab_model.pyx (hunk 2 above)
Comment (MichaelClerx) [reply fc/templates/weblab_model.pyx:302]: Just that I'm not sure how to do it in cython

File: fc/protocol.py (hunk 1 above)
Comment (MichaelClerx) [reply fc/protocol.py:667]: I'm not sure!

Happy to drop this PR and issue if we prefer the old way.

I'm guessing you could write a better log handler in the `fc-runner` that splits the logged messages into files according to their source

Comment (jonc125) [reply fc/protocol.py:667]: Problem is you've lost the info by the time it gets to the fc-runner.

I was anticipating keeping the `log_progress` etc methods but using the logging library inside them?

Comment (MichaelClerx) [reply fc/protocol.py:667]: Yeah I thought about that but then I'm not sure we're using the logging framework in a sensible way.

As far as I understand the idea is that log everything to some central point, with both a logging level and some identifier for the messages' origin. Whatever happens with the logged messages, including the formatting, is then the responsibility of a handler.

So if we're going to impose our own structure / formatting on top of that, I don't really see the point in using it in the first place

**Comment by MichaelClerx** [comment]
Closing this, don't think it's useful anymore

**Pull request closed without merging**